### PR TITLE
NUMERIC_FAULT error

### PR DESCRIPTION
**Pull request by Caranell**
## Dework Task
https://app.dework.xyz/nation3/app-2?taskId=a30b52d5-e12b-4ebe-8f3f-664aa84a158a

## Related GitHub Issue
Issue #94 

## How Has This Been Tested?

Try to use inputs on `liquidity` or `lock` page, entering numbers with more than 18 decimals. The value should stop changing after 18 decimals

### CODE REVIEW

File: ui/components/EthersInput.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+
+export default function EthersInput({ onChange, ...props}: any) {
+    const onInputChange = (e: any) => {
+        const value = e.target.value
+        const isValid = value.toString().match(/^(\d*\.{0,1}\d{0,18}$)/)
```
Comment (johnmark13) [inline ui/components/EthersInput.tsx:6]: I really like this, is everyone else comfortable with regular expressions? As Aahna is adding Jest, this is the kind of thing that should have a unit test wrapped around it (with a lot of cases), but it works and it is neat 👍

Comment (aahna-ashina) [reply ui/components/EthersInput.tsx:6]: @Caranell Now that #96 has been merged, could you add a few unit tests of this function.  That way we can see some human-readable examples valid/invalid values 😄

**Review by aahna-ashina — CHANGES_REQUESTED** [review]

File: ui/components/EthersInput.tsx (hunk 1 above)
Comment (Caranell) [reply ui/components/EthersInput.tsx]: Sure! Added tests

**Review by aahna-ashina — CHANGES_REQUESTED** [review]
@Caranell Typing a number into the _unstake_ input field not longer works:  https://app-8jeucme1s-nation3.vercel.app/liquidity

> Screen Shot 2022-08-29 at 10 09 22 AM

**Comment by Caranell** [comment]
Fixed. Sorry, had problems with the laptop so couldn't test everything properly

**Review by johnmark13 — APPROVED** [review]

**Comment by aahna-ashina** [comment]
@Caranell I added an integration test for trying to type more than 18 decimals:  https://github.com/nation3/app/pull/97/commits/dd42533371ae2ec3d46171975fea0d26529c1261

**Comment by xPi2** [comment]
I can type more than 18 decimals in the preview but it doesn't crash and seems to introduce a valid 18 decimals value in the transaction, is this the expected behaviour? Looks goot to me

Screenshot 2022-09-03 at 13 00 10

**Review by xPi2 — APPROVED** [review]

**Review by aahna-ashina — APPROVED** [review]
Thank you, @Caranell.  Great work! 🙌

**Comment by aahna-ashina** [comment]
> I can type more than 18 decimals in the preview but it doesn't crash and seems to introduce a valid 18 decimals value in the transaction, is this the expected behaviour? Looks goot to me
> 
> Screenshot 2022-09-03 at 13 00 10
> 

@xPi2 The number you typed in the screenshot has zero decimals (comma instead of period).  What was the resulting value in the transaction?

**Comment by xPi2** [comment]
> @xPi2 The number you typed in the screenshot has zero decimals (comma instead of period)? What was the resulting value you ended up with in the transaction? Is there an Etherscan URL we can look at for debugging?

It works exactly the same if you use `,` or `.`. I was also testing that and I didn't realized in the screenshot, I think it works because is parsed on the hook.

The behaviour for me is as follow:
1. I type a decimal number with `.` and continue typing until I reach 18 decimals.
2. After 18 decimals the UI makes a quick glitch and convert the `.` to a `,` an lets me continue adding numbers but only taking into account the first 18 decimal positions.

Good enough for me

**Pull request merged**